yt_dlp/extractor/netverse.py

```diff
@@ -141,27 +157,56 @@ def _real_extract(self, url):
 
 class NetversePlaylistIE(NetverseBaseIE):
     _VALID_URL = r'https?://(?:\w+\.)?netverse\.id/(?P<type>webseries)/(?P<display_id>[^/?#&]+)'
-    _TEST = {
+    _TESTS = [{
+        # multiple season
         'url': 'https://netverse.id/webseries/tetangga-masa-gitu',
         'info_dict': {
             'id': 'tetangga-masa-gitu',
             'title': 'Tetangga Masa Gitu',
         },
-        'playlist_count': 46,
-    }
+        'playlist_count': 519,
+    }, {
+        # single season
+        'url': 'https://netverse.id/webseries/kelas-internasional',
+        'info_dict': {
+            'id': 'kelas-internasional',
+            'title': 'Kelas Internasional',
+        },
+        'playlist_count': 158,
+    }]
 
-    def parse_playlist(self, url, page_num):
-        _, playlist_json = self._call_api(url, query={'page': page_num})
-        for slug in traverse_obj(playlist_json, ('response', 'related', 'data', ..., 'slug')):
+    def parse_single_season_playlist(self, input_data, page_num, custom_id=None, season_id='',
+                                     force_endpoint_type='auto', input_type='url'):
+
+        _, playlist_json = self._call_api(
+            input_data, query={'page': page_num + 1}, season=season_id, custom_id=custom_id,
+            force_endpoint_type=force_endpoint_type, input_type=input_type)
+        for slug in traverse_obj(playlist_json, ('response', ..., 'data', ..., 'slug')):
             yield self.url_result(f'https://www.netverse.id/video/{slug}', NetverseIE)
 
+    def parse_playlist(self, url, json_data, playlist_id):
+        slug_sample = traverse_obj(json_data, ('related', 'data', ..., 'slug'))[0]
+        season_id_list = [season.get('id') for season in json_data.get('seasons')]
+
+        for season in season_id_list:
+            # initial data
+            _, playlist_json = self._call_api(
+                input_data=slug_sample, custom_id=playlist_id, season=season,
+                force_endpoint_type='season', input_type='slug')
+
+            season_list = traverse_obj(playlist_json, ('response', 'season_list'))
+            number_video_per_page = season_list.get('to') - season_list.get('from') + 1
+            number_of_pages = season_list.get('last_page')
+
+            yield from InAdvancePagedList(
+                functools.partial(
+                    self.parse_single_season_playlist, slug_sample, custom_id=playlist_id,
+                    season_id=season, force_endpoint_type='season', input_type='slug'),
+                number_of_pages, number_video_per_page)
```

Comment:
What's the point of pagedlist if you are yielding from it??

Reply:
Actually i want to make the function more readable as the season video basically nested playlist (they didn't give another season in separate link ) but i will change code if it look bad

Reply:
I change this a bit different in https://github.com/yt-dlp/yt-dlp/pull/3854/commits/7d146a007a8c593b1a9d6d36effff4397c5e64dd, let me know if this can be improved